internal/middleware/account.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+)
+
+func AccountMiddleware(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		rhId := ctxval.GetIdentity(r.Context())
+		accDao, err := dao.GetAccountDao(r.Context())
+		if err != nil {
+			ctxval.GetLogger(r.Context()).Error().Err(err).Msg("Failed to initialize connection to fetch Account info")
+			// TODO should we fail hard?
```

Comment:
Wondering why not? what needs to be considered here?

Reply:
Probably nothing, I just will need to discuss how we fail in middlewares :)

Reply:
Absolutely, if DB is down or there is a SQL syntax error, we simply cannot continue - that is 500 and call us to fix this.

Reply:
Uh these multiple `GetLogger` calls are bugging me, but again, these are error states. If this was in the success code path, I would be against it but... lucky you :)